http-gateway/web/src/common/services/ws-manager.js

```diff
@@ -0,0 +1,206 @@
+      this.ws[name].onMessage = () => {}
+      this.ws[name].onClose = () => {}
+      this.ws[name].onOpen = () => {}
+      this.ws[name].onError = () => {}
+      this.ws[name].disconnect()
+      delete this.ws[name]
+    }
+  }
+
+  /**
+   * Add a new connection to the WS Clients.
+   * registerWSClients will be called after that,
+   * which sets all the events and callbacks needed,
+   * and triggering .connect()
+   * @param { name: String, api: String, listener: Function, delayMessage: integer } config
+   * name - Name of the WS client (unique WS instance key) - Required
+   * api - WS endpoint url - Required
+   * listener - WS listener (onMessage)
+   * delayMessage - Delay time for registering the listener
+   */
+  addWsClient = ({
+    name,
+    api,
+    listener = null,
+    delayMessage = this.defaultDelayMessageTimeMs,
+    ...rest
+  }) => {
+    if (name && api) {
+      this.wsClientList = {
+        ...this.wsClientList,
+        [name]: {
+          api,
+          listener,
+          delayMessage,
+          ...rest,
+        },
+      }
+
+      this.registerWSClients()
+    }
+  }
+
+  /**
+   * Removes a WS client.
+   * @param {String} name - Name of the WS client.
+   */
+  removeWsClient = name => {
+    this._unregisterWS(name)
+
+    if (this.wsClientList[name]) {
+      delete this.wsClientList[name]
+    }
+  }
+
+  /**
+   * Removes all WS clients which includes the name (or part of the name) given in the partialName argument.
+   * @param {String} partialName - Partial or full name of the WS client.
+   */
+  removeAllByPartialNameFromWsClient = partialName => {
+    Object.keys(this.ws)
+      .filter(key => key.indexOf(partialName) !== -1)
+      .forEach(key => this.removeWsClient(key))
+  }
+
+  /**
+   * Register all WebsocketClients, which are currently present in the wsClientList.
+   * Already registered clients will be ignored.
+   */
+  registerWSClients = () => {
+    // Register clients
+    Object.keys(this.wsClientList).forEach(id => {
+      // Register only if not registered already.
+      if (!this.ws[id]) {
+        const {
+          api,
+          listener,
+          delayMessage = this.defaultDelayMessageTimeMs,
+          onOpen = null,
+          onError = null,
+        } = this.wsClientList[id]
+
+        // Reset the reconnect attempts on register
+        let reconnectAttempts = 0
+
+        // Register the WebsocketClient
+        this._registerWS(id, api)
+
+        // Connect to the WS and Register listeners
+        if (this.ws[id] && listener) {
+          this.ws[id].connect(delayMessage)
+          this.ws[id].onMessage = listener
+
+          // Reconnect on close after "this.reconnectTimeMs" seconds
+          this.ws[id].onClose = () => {
+            if (getAppMode() !== 'production') {
+              console.info(
+                `ws [${id}] was closed, reconnecting in 2 seconds %c@ ${new Date().toUTCString()}`,
+                'color: #255897;'

```

Comment:
will this color be a common constant?

Reply:
Its just a color for the console.log and only used in development, nothing special.